main.js

```diff
@@ -1,14 +1,7 @@
-const aboutSection = document.getElementById('about');
-const scrollToTopButton = document.querySelector('.aside__home-btn');
+const scrollToTopBtn = document.querySelector('.aside__home-btn');
 
-document.addEventListener('scroll', (e) => {
-
-  if (window.scrollY >= aboutSection.getBoundingClientRect().top) {
-    scrollToTopButton.classList.add('visible');
-  } else {
-    scrollToTopButton.classList.remove('visible');
-  }
-
-  
-}
-)
\ No newline at end of file
+window.addEventListener('scroll', () => {
+  window.scrollY > window.innerHeight / 2 ?
+  scrollToTopBtn.classList.add('active') :
+  scrollToTopBtn.classList.remove('active');
```

Comment:
Lepszą wersją, bardziej "mówiącą co tutaj się dzieje" będzie:
```
if (window.scrollY > window.innerHeight / 2) {
  scrollToTopBtn.classList.add('active') :
} else {
  scrollToTopBtn.classList.remove('active');
}
```
`if (...) {...} else {...}` właśnie bardziej daje "wrażenie" jakiś działań, operacji.


Wyrażenie warunkowe jest bardziej po to aby, warunkować wartość zmiennej czy wyniku funkcji: np. 
```
 function coJeKrowa(tempertatura) {
   return temperature > 0 ? 'trawa' : 'siano';
}
```
co jest ładniejsze i bardziej sensowane niż `if (...) {...} else {...}`

Reply:
W ogóle tutaj możemy zaszaleć z przerobieniem tego na rożne sposoby, będzie to nawet **przesadzone**, ale będziesz mógł coś z tego wynieść w kontekście kodu i jego reużywalności, czytelności i nie powtarzania się:

```
const ACTIVE_MODIFIER = 'active`; // odrazu nie powtarzasz się, a gdybyś nagle zmienił zdanie co to nazwy klasy to możesz to robisz to w jednym miejscu

window.addEventListener('scroll', () => {
  if (window.scrollY > window.innerHeight / 2) {
    scrollToTopBtn.classList.add(ACTIVE_MODIFIER) :
  } else {
    scrollToTopBtn.classList.remove(ACTIVE_MODIFIER);
  }
})
```

Możemy jeszcze bardziej zaszaleć, co zdecydowanie będzie przesadą:
```
window.addEventListener('scroll', () => {
  const action = window.scrollY > window.innerHeight / 2 ? 'add' : 'remove'; 
  scrollToTopBtn.classList[action]('active');
})
```
Potraktuj to tylko jako ćwiczenie myślowe.